teuthology/task/install/__init__.py

```diff
@@ -191,15 +191,17 @@ def get_package_list(ctx, config):
         debs = filter(lambda p: not p.endswith('-dbg'), debs)
         rpms = filter(lambda p: not p.endswith('-debuginfo'), rpms)
 
-    excluded_packages = config.get('exclude_packages', list())
-    if excluded_packages:
-        log.debug("Excluding packages: {}".format(excluded_packages))
-
-        def exclude(pkgs):
-            return list(set(pkgs).difference(set(excluded_packages)))
+    def exclude(pkgs, exclude_list):
+        return list(set(pkgs).difference(set(exclude_list)))
 
-        debs = exclude(debs)
-        rpms = exclude(rpms)
+    excluded_packages = config.get('exclude_packages', [])
+    if isinstance(excluded_packages, dict):
```

Comment:
Here and after, I would suggest to use utility method for getting packages:
```
def getpkg(_dict, _key, _type):
  if _key in _dict:
    return _dict[_key].get(_type, []) if isinstance(_dict[_key], dict) else _dict[_key] or []
  else:
    return []
debs = exclude(debs, getpkg(config, 'exclude_packages', 'deb')
rpms = exclude(rpms, getpkg(config, 'exclude_packages', 'rpm')
.
.
.
debs += getpkg(config, 'extra_packages', 'deb')
rpms += getpkg(config, 'extra_packages, 'rpm')
.
.
.
pkgs += getpkg(config, 'extra_packages', system_type)
.
.
.
system_pkglist = getpkg(config, 'extra_system_packages', 'deb')
```

Reply:
This is a backport, so I'll stick with the upstream code. Maybe you just want to open an upstream PR with your desired changes?

Reply:
@smithfarm You're right